### PR TITLE
Remove PY3 variable

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: IPython/core/tests/test_splitinput.py [hunk 1]
```diff
@@ -28,10 +28,7 @@
     ('??%%hist4', ('', '??', '%%hist4', '')),
     ('?x*', ('', '?', 'x*', '')),
     ]
-if py3compat.PY3:
     tests.append((u"Pérez Fernando", (u'', u'', u'Pérez', u'Fernando')))
```
Comment (takluyver) [inline IPython/core/tests/test_splitinput.py]: This line will need to be de-dented.

File: IPython/core/ultratb.py [hunk 2]
```diff
@@ -315,16 +315,6 @@ def wrapped(*args, **kwargs):
...
 
 
-if py3compat.PY3:
-    fixed_getargvalues = inspect.getargvalues
-else:
-    # Fixes for https://github.com/ipython/ipython/issues/8293
-    #       and https://github.com/ipython/ipython/issues/8205.
-    # The relevant bug is caused by failure to correctly handle anonymous tuple
-    # unpacking, which only exists in Python 2.
-    fixed_getargvalues = with_patch_inspect(inspect.getargvalues)
```
Comment (takluyver) [inline IPython/core/ultratb.py]: Is the `with_patch_inspect` function still used anywhere? We might want to mark it as deprecated.

Comment (srinivasreddy) [inline IPython/core/ultratb.py]: Nope. https://github.com/ipython/ipython/search?utf8=%E2%9C%93&q=with_patch_inspect

Comment (takluyver) [inline IPython/core/ultratb.py]: OK, then can you add a note to its docstring saying that it's deprecated since IPython 6.0?

Comment (srinivasreddy) [inline IPython/core/ultratb.py]: sure.

Comment (srinivasreddy) [inline IPython/core/ultratb.py]: Done.

**Comment by takluyver** [comment]
This is getting really large again; be aware that large pull requests are likely to be waiting longer for review.

**Comment by srinivasreddy** [comment]
Okay. I will stop pushing this branch. :smile:

File: IPython/core/ultratb.py [hunk 3]
```diff
@@ -299,7 +299,10 @@ def getargs(co):
 
 # Monkeypatch inspect to apply our bugfix.
 def with_patch_inspect(f):
-    """decorator for monkeypatching inspect.findsource"""
+    """
+    Deprecated since IPython 6.0
+    decorator for monkeypatching inspect.findsource
+    """
```
Comment (Carreau) [inline IPython/core/ultratb.py]: We may want to raise a deprecation warning as well, but that can be done in a subsequent PR.

File: IPython/utils/ulinecache.py [hunk 4]
```diff
@@ -10,35 +8,9 @@
 from IPython.utils import py3compat
 from IPython.utils import openpy
 
-if py3compat.PY3:
-    getline = linecache.getline
-    
-    # getlines has to be looked up at runtime, because doctests monkeypatch it.
-    @functools.wraps(linecache.getlines)
-    def getlines(filename, module_globals=None):
-        return linecache.getlines(filename, module_globals=module_globals)
```
Comment (Carreau) [inline IPython/utils/ulinecache.py]: that can likely be reduced further, as now the wrapper is no-op, we can just search for the usage, and use `linecache.getlines` directly. We can leave this function there with a deprecation warning.

**Review by Carreau — APPROVED** [review]
A couple of things that can be reduced/improved further but that can be done separately later.

**Comment by Carreau** [comment]
I rebased to resolve conflict with master. Waiting for test to pass.

File: IPython/utils/ulinecache.py (hunk 4 above)
Comment (srinivasreddy) [inline IPython/utils/ulinecache.py]: Done. Please take a look. @Carreau

**Comment by srinivasreddy** [comment]
@Carreau  Let's get this merged too.

**Pull request merged**